File: 0046-permutations/0046-permutations.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> list=new ArrayList<>();
        perm(nums,0,list);
        return list;
    }
    public void perm(int[]nums,int fi,List<List<Integer>> list)
    {
        if(fi==nums.length)
        {
            List<Integer> temp=new ArrayList<Integer>();
            for (int i:nums)
                temp.add(i);
            list.add(temp);
            return ;
        }

        for (int i=fi;i<nums.length;i++)
        {
            int t=nums[fi];
            nums[fi]=nums[i];
            nums[i]=t;
            perm(nums,fi+1,list);
             t=nums[fi];
            nums[fi]=nums[i];
            nums[i]=t;

        }
    }

}

File: 0046-permutations/0046-permutations.py
class Solution:
    def find(self,nums,fix,res):
        if fix==len(nums)-1:
            res.append(nums[ : ])
            return
        for i in range(fix,len( nums)):
            nums[i], nums[fix]=nums[fix],nums[i]
            self.find(nums, fix+1,res)
            nums[i], nums[fix]=nums[fix],nums[i]

    def permute(self, nums: List[int]) -> List[List[int]]:
        res=[]
        self.find(nums,0,res)
        return res

        